craft_store/base_client.py

```diff
@@ -316,3 +316,36 @@ def list_registered_names(
         response = self.request("GET", self._base_url + endpoint, params=params)
         results = response.json().get("results", [])
         return [models.RegisteredNameModel.unmarshal(item) for item in results]
+
+    def register_name(
+        self,
+        name: str,
+        *,
+        entity_type: Optional[Literal["charm", "bundle", "snap"]] = None,
+        private: bool = False,
+        team: Optional[str] = None,
+    ) -> str:
+        """Register a name on the store.
+
+        :param name: the name to register.
+        :param entity_type: The type of package to register (e.g. charm or snap)
+        :param private: Whether this entity is private or not.
+        :param team: An optional team ID to register the name with.
+
+        :returns: the ID of the registered name.
+
+        Further Detail: https://api.charmhub.io/docs/default.html#register_name
```

Comment:
can you find out if the store team if there is a more generic url for this?

Reply:
I've asked, and in the meantime I've removed this. I'll make a follow-up PR for an appropriate doc link if there is one.